File: src/main/scala/interview/scala/rmsolutions/LC167TwoSumIIInputarrayissorted.scala
package interview.scala.rmsolutions

/** [[https://leetcode.com/problems/two-sum-ii-input-array-is-sorted]] */
class LC167TwoSumIIInputarrayissorted {
  def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
    var low: Int = 0
    var high: Int = numbers.length - 1

    while (low < high) {
      val total = numbers.apply(low) + numbers.apply(high)
      if (total == target) {
        return Array(low + 1, high + 1)
      }
      if (total < target) {
        low = low + 1
      } else {
        high = high - 1
      }
    }

    null
  }
}

File: src/main/java/interview/java/rmsolutions/LC167TwoSumIIInputarrayissorted.java
package interview.java.rmsolutions;

/**
 * @see <a href="https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/">Link</a>
 */
public class LC167TwoSumIIInputarrayissorted {
    public int[] twoSum(int[] numbers, int target) {
        for (int i = 0, j = 1; j < numbers.length && i < numbers.length - 1; ) {
            if (numbers[i] + numbers[j] > target || j < numbers.length - 1) {
                for (int k = i + 1; k < j; k++) {
                    if (numbers[i] + numbers[j] == target) {
                        return new int[]{i + 1, j + 1};
                    }
                    i++;
                }
            }
            if (numbers[i] + numbers[j] == target) {
                return new int[]{i + 1, j + 1};
            }

            if (j < numbers.length - 1) {
                j++;
            }
        }
        return null;
    }
}
